.gitlab-ci.yml

```diff
@@ -0,0 +1,151 @@
+ PIP_EXTRA_INDEX_URL: https://nexus.osdc.io/repository/pypi-all/simple
+ DOCKER_BUILD_OPTS: "--build-arg PIP_EXTRA_INDEX_URL=https://nexus.osdc.io/repository/pypi-all/simple"
+ DOCKER_PUSH_OPTS: "--all-tags"
+ TWINE_REPOSITORY_URL: "https://nexus.osdc.io/repository/pypi-snapshots/"
+
+
+tox:
+  image: ${BASE_CONTAINER_REGISTRY}/${REPO_PY_VERSION}-builder:${BASE_CONTAINER_VERSION}
+  before_script:
+    - mkdir -p /usr/share/man/man1
+
+.python_versions:
+  parallel:
+    matrix:
+      - REPO_PY_VERSION: ["python36"]
+
+docker_build:
+  allow_failure: true
+  rules:
+    - when: never
+
+release:
+  image: docker.osdc.io/ncigdc/python36-builder:1.4.0
+  allow_failure: true
+  rules:
+    - when: never
+
+update_reqs_for_gdcdatamodel2:
+  image: docker.osdc.io/ncigdc/python36-builder:1.4.0
+  stage: build_gdcdatamodel2
+  script:
+    - !reference [.load_github_key, script]
+    - git clone git@github.com:NCI-GDC/gdcdatamodel2.git
+    - cd gdcdatamodel2
+    - git checkout "$CI_COMMIT_REF_NAME" || git checkout -b "$CI_COMMIT_REF_NAME"
+    - git log -1
+    - sed -i 's|gdcdictionary @ git+https://github.com/NCI-GDC/gdcdictionary.git@.*#egg=gdcdictionary|gdcdictionary @ git+https://github.com/NCI-GDC/gdcdictionary.git@'"$CI_COMMIT_REF_NAME"'#egg=gdcdictionary|' setup.cfg
+    - cat setup.cfg
+    - pip3 --version
+    - pip3 install virtualenv pip-tools==6.3.0
+    - pip-compile -i https://pypi.org/simple --extra=dev --output-file=dev-requirements.txt pyproject.toml
+  artifacts:
+    paths:
+      - gdcdatamodel2/setup.cfg
+      - gdcdatamodel2/dev-requirements.txt
+    expire_in: 1 week
+
+
+update_datamodels:
+  image: docker.osdc.io/ncigdc/python3.8-builder:2.1.0
+  stage: build_gdcdatamodel2
+  script:
+    - !reference [.load_github_key, script]
+    - pwd
+    - git clone git@github.com:NCI-GDC/gdcdatamodel2.git
+    - cd gdcdatamodel2
+    - git checkout "$CI_COMMIT_REF_NAME" || git checkout -b "$CI_COMMIT_REF_NAME"
+    - git log -1
+    - sed -i 's|version = .*|version ="'"$CI_COMMIT_REF_NAME"'"|' plaster.toml
+    - sed -i 's/is_tag = .*/is_tag = false/' plaster.toml
+    - cat plaster.toml
+    - bash plaster
+    - git status
+    - rm setup.cfg dev-requirements.txt
+  artifacts:
+    paths:
+      - gdcdatamodel2
+    expire_in: 1 week
+
+
+release_datamodels_to_nexus:
+  image: docker.osdc.io/ncigdc/python36-builder:1.4.0
+  stage: release_gdcdatamodel2
+  script:
+    - cd gdcdatamodel2
+    - ls -R .
+    - cat setup.cfg
+    - cat dev-requirements.txt
+    - whoami
+    - apt-get update
+    - apt-get install -y python3-venv
+    - pip3 install -U setuptools_scm build twine setuptools
+    - cp -v pyproject.toml /tmp/pyproject.toml
+    - sed -i 's/local_scheme = "no-local-version"/local_scheme = "node-and-timestamp"/' pyproject.toml
+    - cat pyproject.toml
+    - python3 -m setuptools_scm
+    - python3 -m build
+    - twine check dist/*
+    - twine upload dist/*
+    - cp -v /tmp/pyproject.toml pyproject.toml
```

Comment:
As improvement for later, we can make use of the tox:release command defined the gdcdatamodel2's tox.ini